src/systems/model_photo_shoot/ModelPhotoShoot.hh

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (C) 2018 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+#ifndef IGNITION_GAZEBO_SYSTEMS_MODELPHOTOSHOOT_HH_
+#define IGNITION_GAZEBO_SYSTEMS_MODELPHOTOSHOOT_HH_
+
+#include <sdf/sdf.hh>
+
+#include <memory>
+
+#include <ignition/gazebo/Model.hh>
+#include <ignition/gazebo/System.hh>
+#include <ignition/gazebo/World.hh>
+#include <ignition/rendering/Camera.hh>
+#include <ignition/rendering/RenderingIface.hh>
+#include <ignition/transport/Node.hh>
+
+namespace ignition
+{
+namespace gazebo
+{
+// Inline bracket to help doxygen filtering.
+inline namespace IGNITION_GAZEBO_VERSION_NAMESPACE {
+namespace systems
+{
+  // Forward declarations.
+  class ModelPhotoShootPrivate;
+
+  /// \brief This plugin is a port of the old ModelPropShop plugin from gazebo
+  /// classic. It generates 5 pictures of a model: perspective, top, front,
+  /// sideand back. It can do it using the default position or moving the joint
+  /// to random positions. It allows saving the camera and joint poses so it
+  /// can be replicated in other systems.
+  ///
+  /// ## System Parameters
+  /// - <translation_data_file> - Location to save the camera and joint poses.
+  ///   [Optional]
+  /// - <random_joints_pose> - Set to true to take pictures with the joints in
+  ///   random poses instead of the default ones. This option only supports
+  ///   single axis joints. [Optional]
+  ///
+  /// ## Example
+  /// An example configuration is installed with Gazebo. The example uses
+  /// the Ogre2 rendering plugin to set the background color of the pictures.
+  /// It also includes the camera sensor that will be used along with the
```

Comment:
This camera is not optional right? From what I can tell looking at the code, if I want to use this plugin, I have to have a camera with a specific name. Do you mind adding some detail about this?

Reply:
Yes, I was planning to add it to the tutorial, but now that you say that I agree it might be worth mentioning on the `.hh` too.